website/docs/r/arc_machine_automanage_configuration_assignment.html.markdown

```diff
@@ -0,0 +1,78 @@
+---
+subcategory: "Automanage"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_arc_machine_automanage_configuration_assignment"
+description: |-
+  Manages an Arc Machine Automanage Configuration Profile Assignment.
+---
+
+# azurerm_arc_machine_automanage_configuration_assignment
+
+Manages an Arc Machine Automanage Configuration Profile Assignment.
+
+## Example Usage
+
+```hcl
+provider "azurerm" {
+  features {}
+}
+
+variable "arc_machine_name" {
+  description = "The name of the Arc Machine."
+}
+
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+data "azurerm_arc_machine" "example" {
+  name                = var.arc_machine_name
+  resource_group_name = azurerm_resource_group.example.name
+}
+
+resource "azurerm_automanage_configuration" "example" {
+  name                = "example-configuration"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+}
+
+resource "azurerm_arc_machine_automanage_configuration_assignment" "example" {
+  arc_machine_id   = data.azurerm_arc_machine.example.id
+  configuration_id = azurerm_automanage_configuration.example.id
```

Comment:
should this be
```suggestion
  automanage_configuration_id = azurerm_automanage_configuration.example.id
```
so its clear what configuration resource type is expected?

Reply:
Hi @katbyte, I opted for "configuration_id" over "automanage_configuration_id" because the name of the resource already implies it's an automanage configuration. This choice also keeps it consistent with other resources, like [azurerm_virtual_machine_automanage_configuration_assignment](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/virtual_machine_automanage_configuration_assignment#configuration_id). However, I'm open to switching to "automanage_configuration_id" if you believe it's necessary.